include/aws/common/thread.h

```diff
@@ -110,10 +112,16 @@ AWS_COMMON_API
 void aws_thread_clean_up(struct aws_thread *thread);
 
 /**
- * returns the thread id of the calling thread.
+ * Returns the thread id of the calling thread.
+ */
+AWS_COMMON_API
+aws_thread_id aws_thread_current_thread_id(void);
```

Comment:
This can't fail. On posix platforms it maps to `pthread_self()`. On windows to `GetCurrentThreadId()`.

http://man7.org/linux/man-pages/man3/pthread_self.3.html
https://docs.microsoft.com/en-us/windows/win32/api/processthreadsapi/nf-processthreadsapi-getcurrentthreadid